feature/home/src/main/java/com/epmedu/animeal/home/presentation/viewmodel/HomeViewModel.kt

```diff
@@ -35,15 +37,21 @@ internal class HomeViewModel @Inject constructor(
 
         viewModelScope.launch {
             locationProvider.fetchUpdates().collect {
-                updateState { copy(currentLocation = it) }
+                updateState { copy(currentLocation = MapLocation(it)) }
             }
         }
 
         viewModelScope.launch {
-            gpsSettingsProvider.fetchUpdates().collect {
-                updateState { copy(gpsSettingState = it) }
-            }
+            gpsSettingsProvider.fetchUpdates().collect(::collectGpsSettings)
+        }
+    }
+
+    private fun collectGpsSettings(state: GpsSettingsProvider.GpsSettingState) {
+        val uiGpsState = when (state) {
+            GpsSettingsProvider.GpsSettingState.Enabled -> GpsSettingState.Enabled
```

Comment:
can be just `GpsSettingState.Enabled`

Reply:
No-no-no, there are two different GpsSettingState. One from GpsSettingsProvider(model) and other just GpsSettingState(presenter). In other case we should write @Stable annotation. Here our ui and model are independent.